base/compiler/typeutils.jl

```diff
@@ -5,7 +5,7 @@
 #####################
 
 function rewrap(@nospecialize(t), @nospecialize(u))
-    if isa(t, TypeVar) || isa(t, Type)
+    if isa(t, TypeVar) || isa(t, Type) || isa(t, Core.TypeofVararg)
```

Comment:
I did add a specialization of `rewrap_unionall` that handles this case (since anything else would be breaking), but I'm not all that convinced that `rewrap_unionall` should actually work on Vararg. In many cases it's the wrong thing to do. Maybe just make the `Vararg` case explicit here, since this is an internal function? Alternatively, I think there's only one or two callsites of this that allow a Vararg, so maybe they should use a different entrypoint.

Reply:
I think the || already makes it pretty explicit. We can consider other designs later as continued clean up, but this seems to fix CI regression for now.

Reply:
alright